modules/hypothesis.py

```diff
@@ -15,20 +17,75 @@ def get_items(self) -> Mirror.Results:
                 continue
             yield node(
                 heading=dt_heading(page.created, link(title=page.title, url=page.url)),
-                children=[node(
-                    heading=dt_heading(hl.created, link(title='context', url=hl.hyp_link)),
-                    tags=hl.tags,
-                    body=hl.highlight,
-                    children=[] if hl.annotation is None else [node(
-                        heading=hl.annotation,
-                    )]
-                ) for hl in page.highlights]
+                properties=self.get_page_properties(page),
+                children=[
+                    node(
+                        heading=dt_heading(
+                            hl.created, link(title="context", url=hl.hyp_link)
+                        ),
+                        properties=self.get_annotation_properties(hl),
+                        tags=hl.tags,
+                        body=hl.highlight,
+                        children=[]
+                        if hl.annotation is None
+                        else [
+                            node(
+                                heading=self.get_annotation(hl.annotation).split("\n")[
+                                    0
+                                ],
+                                body=self.get_annotation(hl.annotation),
+                            )
+                        ],
+                    )
+                    for hl in page.highlights
+                ],
             )
 
+    def get_annotation(self, annotation: Optional[str]) -> Optional[str]:
+        assert self.cmdline_args is not None
+        if not self.cmdline_args.markdown:
+            return annotation
+
+        from orger import pandoc
+
+        org = pandoc.to_org(str(annotation), from_="markdown")
+        return org
+
+    def get_annotation_properties(
+        self, highlight: hypothesis.Highlight
+    ) -> Optional[Mapping[str, str]]:
+        assert self.cmdline_args is not None
+        if not self.cmdline_args.id_per_annotation:
+            return None
+
+        return {"ID": str(uuid.uuid3(uuid.NAMESPACE_URL, highlight.hyp_link))}
+
+    def get_page_properties(self, page: hypothesis.Page) -> Optional[Mapping[str, str]]:
+        assert self.cmdline_args is not None
+        if not self.cmdline_args.id_per_page:
+            return None
+
+        return {"ID": str(uuid.uuid3(uuid.NAMESPACE_URL, page.url))}
+
 
 # TODO tests for determinism
 # TODO they could also be extracted to common routine and used in each provider
 # TODO need to group by source??
 
-if __name__ == '__main__':
-    HypView.main()
+
+def setup_parser(p) -> None:
+    p.add_argument(
+        "--markdown",
```

Comment:
actually I don't think you need this, IMO it makes sense to do by default, and if someone wants to disable it there is a `--disable-pandoc` argument
https://github.com/karlicoss/orger/blob/9094db398341b0354fc51abb344e21fad8980949/src/orger/org_view.py#L72-L76

and then `to_org` will automatically respect it, so no need to check in `get_annotation` either
```
def to_org(data: str, *, from_: str, logger=logging) -> str:
    if not should_use_pandoc():
        return data
```